### DISCUSSION TITLE
Column data encrypted with pgp_sym_encrypt; using realtime - is there a way to decrypt the value of the column?

### DISCUSSION THREAD

**Discussion by pandisFox** [supabase/supabase#25160] [Questions] [answered]
Hey all,

I have a column which stores the result of calling `pgp_sym_encrypt`* but as expected it gets broadcasted encrypted in the client, is there a way around this so I can display the decrypted value?

I know that upon getting the realtime notification I can call a function to give me the decrypted value:

`select pgp_sym_decrypt(message::bytea, 'secret_key') as message` - this gives me back the value decrypted - I can create a function around it, but is there a different way to handle it? 

Thanks  
  
  
*specifically, inserting the result of calling: `pgp_sym_encrypt (message, 'secret_key')`

**Comment by GaryAustin1** [comment, marked as answer]
You cannot join or use views with realtime, so no.

**Reply by pandisFox** [reply]
thanks!
